### PR TITLE
Prototype functionality to add perspective saving.

### PR DESCRIPTION
**Pull request by willrogers**
For further discussion @nickbattam 

Requires Mars 4.5.2.

### CODE REVIEW

File: applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/src/org/csstudio/opibuilder/runmode/OPIView.java [hunk 1]
```diff
@@ -119,9 +126,12 @@ public void init(final IViewSite site, final IMemento memento) throws PartInitEx
...
                                (memento == null ? ", no memento" : "with memento"));
 
-        if (memento == null)
+        if (memento == null) {
+            memento = findMementoFromPlaceholder();
+        }
+        if (memento == null) {
             return;
-
+        }
```
Comment (nickbattam) [inline applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/src/org/csstudio/opibuilder/runmode/OPIView.java]: tab

File: applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/src/org/csstudio/opibuilder/runmode/OPIView.java [hunk 2]
```diff
@@ -139,8 +149,64 @@ public void init(final IViewSite site, final IMemento memento) throws PartInitEx
         final IAdaptable element = factory.createElement(inputMem);
         if (!(element instanceof IEditorInput))
             throw new PartInitException("Instead of IEditorInput, " + factoryID + " returned " + element);
+        input = (IEditorInput) element;
+
         // Set input, but don't persist to memento because we just read it from memento
-        setOPIInput((IEditorInput)element, false);
+        setOPIInput((IEditorInput)input, false);
```
Comment (nickbattam) [inline applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/src/org/csstudio/opibuilder/runmode/OPIView.java]: no need for cast here

File: core/ui/ui-plugins/org.csstudio.perspectives/src/org/csstudio/perspectives/PerspectiveSaver.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            try {
+                MPerspective p = (MPerspective) o;
+
+                List<MPlaceholder> phs = modelService.findElements(p, null, MPlaceholder.class, null);
+                for (MPlaceholder ph : phs) {
+                    ph.getPersistedState()
+                            .putAll(ph.getRef().getPersistedState());
+                }
+                MPerspective clone = (MPerspective) modelService.cloneElement(p, null);
+                System.out.println("URL: " + url);
```
Comment (nickbattam) [inline core/ui/ui-plugins/org.csstudio.perspectives/src/org/csstudio/perspectives/PerspectiveSaver.java]: log not print

File: applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/src/org/csstudio/opibuilder/runmode/OPIView.java (hunk 1 above)
Comment (willrogers) [inline applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/src/org/csstudio/opibuilder/runmode/OPIView.java]: ... no it's not ...

File: core/ui/ui-plugins/org.csstudio.perspectives/src/org/csstudio/perspectives/PerspectiveLoader.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        File dirname = new File(chooser.getFilterPath());
+        File fullPath = new File(dirname, chooser.getFileName());
+        if (fullPath.isFile()) {
+            ResourceSet rs = new ResourceSetImpl();
+            URI uri = URI.createURI(IPerspectiveUtils.FILE_PREFIX + fullPath.getPath());
+            Resource res = rs.getResource(uri, true);
+            EObject obj = res.getContents().get(0);
+            if (obj instanceof MPerspective) {
+                MPerspective p = (MPerspective) obj;
+                MPerspective pclone = (MPerspective) modelService.cloneElement(p, null);
```
Comment (nickbattam) [inline core/ui/ui-plugins/org.csstudio.perspectives/src/org/csstudio/perspectives/PerspectiveLoader.java]: is it necessary to clone before calling perspToString()?

File: applications/pom.xml [hunk 5]
```diff
@@ -271,7 +271,7 @@
   <modules>
     <module>alarm</module>
     <module>archive</module>
-    <module>channel</module>
+    <!-- <module>channel</module> -->
```
Comment (nickbattam) [inline applications/pom.xml]: Should this be deleted?

File: applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/src/org/csstudio/opibuilder/runmode/OPIView.java [hunk 6]
```diff
@@ -119,9 +126,12 @@ public void init(final IViewSite site, final IMemento memento) throws PartInitEx
             System.out.println(site.getId() + ":" + site.getSecondaryId() + " opened " +
```
Comment (nickbattam) [inline applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/src/org/csstudio/opibuilder/runmode/OPIView.java]: debug print

File: core/ui/ui-plugins/org.csstudio.perspectives.test/src/org/csstudio/perspectives/PerspectiveSaverUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+            saver.init();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void createPerspectiveSaver()
+    {
+        System.out.println("Saver " + saver);
```
Comment (nickbattam) [inline core/ui/ui-plugins/org.csstudio.perspectives.test/src/org/csstudio/perspectives/PerspectiveSaverUnitTest.java]: Incomplete test

File: applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/src/org/csstudio/opibuilder/runmode/OPIView.java [hunk 8]
```diff
@@ -119,9 +126,12 @@ public void init(final IViewSite site, final IMemento memento) throws PartInitEx
...
                                (memento == null ? ", no memento" : "with memento"));
```
Comment (nickbattam) [inline applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/src/org/csstudio/opibuilder/runmode/OPIView.java]: debug print

**Pull request merged**